src/app/shared/interfaces/debug-variables.ts

```diff
@@ -0,0 +1,17 @@
+export interface DebugVariables {
+  correlationId?: string;
+  duration?: string;
+  endTime?: string;
+  estimatedMemoryUsage?: string;
+  name?: string;
+  numberOfCheckpoints?: string;
+  status?: string;
+  storageId?: string;
+  storageSize?: string;
+  component?: string;
+  conversationId?: string;
+  endpointName?: string;
+  timestamp?: string;
+
+  [key: string]: string | undefined;
+}
```

Comment:
Can everything be undefined? :o

Reply:
This looks like optional variables meant for debugging.
I guess you could set the default for all of them to false unless a different value is given but this is also fine imo

Reply:
I checked where these variables are used and none of them need to be undefined, so i changed it.

Reply:
Does the API always give those values back?

Reply:
Hm yea, I'm also wondering why there is also a variable key. It might be better to have an `extras` value that includes this 'wildcard'.

Reply:
the [key: string]: string is because the html loops over the variable names to populate the cells

Reply:
If you're going to loop over it anyway, is there even any point in defining the variables? (normally yes, for type safety, but the wildcard defeats the purpose)

Reply:
So why can't it be `Record<string,string>` if you're only using it for looping?
And if you do use some variables, split them up into separate types; one with defined properties and the record to loop over

Reply:
im not sure if i understand what you mean

Reply:
Sorry, i didnt see your comment Matthbo. I can it to be `Record<string, string>`, i dont think that'd be an issue.

Reply:
I have changed it to be `Record<string, string>` now